legend-engine-shared-core/src/test/java/org/finos/legend/engine/shared/core/operational/prometheus/TestErrorManagement.java

```diff
@@ -0,0 +1,427 @@
+import org.finos.legend.engine.shared.core.operational.logs.LoggingEventType;
+import org.junit.After;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.TestCouldNotBeSkippedException;
+import java.util.MissingFormatWidthException;
+import java.util.UnknownFormatFlagsException;
+import static org.finos.legend.engine.shared.core.operational.prometheus.MetricsHandler.toCamelCase;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+public class TestErrorManagement
+{
+    private final String[] ERROR_LABEL_NAMES = {"errorLabel", "category", "source", "serviceName"};
+    private final String METRIC_NAME = "legend_engine_error_total";
+    private final CollectorRegistry METRIC_REGISTRY = MetricsHandler.getMetricsRegistry();
+    private final double DELTA = 0d;
+    private final String TEST_SERVICE_PATH = "service/remote/getRegistry";
+
+    @After()
+    public void clearCounterData()
+    {
+        MetricsHandler.ERROR_COUNTER.clear();
+    }
+
+    @Test
+    public void testServiceErrorOriginLabel()
+    {
+        MetricsHandler.observeError(LoggingEventType.SERVICE_TEST_EXECUTE_ERROR, new Exception(), TEST_SERVICE_PATH);
+        String[] labels = {"ServiceTestExecuteError", "UnknownError", "ServiceTestExecute", TEST_SERVICE_PATH};
+        assertEquals(METRIC_REGISTRY.getSampleValue(METRIC_NAME, ERROR_LABEL_NAMES, labels), 1, DELTA);
+    }
+
+    @Test
+    public void testInteractiveErrorOriginLabel()
+    {
+        MetricsHandler.observeError(LoggingEventType.SERVICE_TEST_EXECUTE_ERROR, new Exception(), null);
+        String[] labels = {"ServiceTestExecuteError", "UnknownError", "ServiceTestExecute", "N/A"};
+        assertEquals(METRIC_REGISTRY.getSampleValue(METRIC_NAME, ERROR_LABEL_NAMES, labels), 1, DELTA);
+    }
+
+    @Test
+    public void testErrorWithoutOrigin()
+    {
+        Exception exception = assertThrows(EngineException.class, () ->
+        {
+            MetricsHandler.observeError(null, new Exception(), TEST_SERVICE_PATH);
+        });
+        assertEquals(exception.getMessage(), "Error origin must not be null!");
+    }
+
+    @Test
+    public void testErrorLabelExtractionWithNonGenericException()
+    {
+        MetricsHandler.observeError(LoggingEventType.CATCH_ALL, new ArithmeticException(), null);
+        String[] labels = {"ArithmeticError", "UnknownError", "CatchAll", "N/A"};
+        assertEquals(METRIC_REGISTRY.getSampleValue(METRIC_NAME, ERROR_LABEL_NAMES, labels), 1, DELTA);
+
+        MetricsHandler.observeError(LoggingEventType.LAMBDA_RETURN_TYPE_ERROR, new NullPointerException(), TEST_SERVICE_PATH);
+        labels = new String[]{"NullPointerError", "UnknownError", "LambdaReturnType", TEST_SERVICE_PATH};
+        assertEquals(METRIC_REGISTRY.getSampleValue(METRIC_NAME, ERROR_LABEL_NAMES, labels), 1, DELTA);
+    }
+
+    @Test
+    public void testErrorLabelExtractionWithEngineExceptionWithType()
```

Comment:
A lot of tests .. which is generally good !

No need to change this now. In the future you might want to consider a Parameterized Junit test https://junit.org/junit4/javadoc/4.12/org/junit/runners/Parameterized.html